src/segments/svn.go

```diff
@@ -0,0 +1,164 @@
+package segments
+
+import (
+	"oh-my-posh/environment"
+	"oh-my-posh/regex"
+	"strconv"
+	"strings"
+	"unicode/utf8"
+)
+
+// SvnStatus represents part of the status of a Svn repository
+type SvnStatus struct {
+	ScmStatus
+}
+
+func (s *SvnStatus) add(code string) {
+	switch code {
+	case "C":
+		s.Conflicted++
+	case "D":
+		s.Deleted++
+	case "A":
+		s.Added++
+	case "M":
+		s.Modified++
+	case "R":
+		s.Moved++
+	default:
+		s.Unmerged++
+	}
+}
+
+func (s *SvnStatus) HasConflicts() bool {
+	return s.Conflicted > 0
+}
+
+type Svn struct {
+	scm
+
+	Working *SvnStatus
+	BaseRev int
+	Branch  string
+
+	svnWorkingFolder string // .Svn working folder
+	svnRootFolder    string // .Svn root folder
+	svnRealFolder    string // .Svn real folder(can be different from current path when in worktrees)
+
+	svnCommand string
+
+	IsWslSharedPath bool
+}
+
+func (s *Svn) Template() string {
+	return " \ue0a0{{.Branch}} r{{.BaseRev}} {{.Working.String}} " // nolint: lll
+}
+
+func (s *Svn) Enabled() bool {
+	if !s.shouldDisplay() {
+		return false
+	}
+	displayStatus := s.props.GetBool(FetchStatus, false)
+	if displayStatus {
+		s.setSvnStatus()
+	} else {
+		s.Working = &SvnStatus{}
+	}
+	return true
+}
+
+func (s *Svn) shouldDisplay() bool {
+	// when in wsl/wsl2 and in a windows shared folder
+	// we must use Svn.exe and convert paths accordingly
+	// for worktrees, stashes, and path to work
+	s.IsWslSharedPath = s.env.InWSLSharedDrive()
+	if !s.env.HasCommand(s.getSvnCommand()) {
+		return false
+	}
+	Svndir, err := s.env.HasParentFilePath(".svn")
+	if err != nil {
+		return false
+	}
+	if s.shouldIgnoreRootRepository(Svndir.ParentFolder) {
+		return false
+	}
+
+	if Svndir.IsDir {
+		s.svnWorkingFolder = Svndir.Path
+		s.svnRootFolder = Svndir.Path
+		// convert the worktree file path to a windows one when in wsl 2 shared folder
+		s.svnRealFolder = strings.TrimSuffix(s.convertToWindowsPath(Svndir.Path), ".svn")
+		return true
+	}
+	// handle worktree
+	s.svnRootFolder = Svndir.Path
+	dirPointer := strings.Trim(s.env.FileContent(Svndir.Path), " \r\n")
+	matches := regex.FindNamedRegexMatch(`^Svndir: (?P<dir>.*)$`, dirPointer)
```

Comment:
we have helpers for this that caches the regex compilations, see `oh-my-posh/regex`